src/universalSort.java

```diff
@@ -0,0 +1,39 @@
+package components.universalSort;
+
+/**
+ * {@code universalSortKernel} enhanced with secondary methods.
+ */
+public interface universalSort extends universalSortKernel {
+
+    /**
+     * sorts {@code this} by the size of the indices, smallest to largest.
+     *
+     * @updates this
+     *
+     * @ensures this = #this
+     */
+    void sortSize();
+
+    /**
+     * sorts {@code this} alphabetally, while accounting for case
+     *
+     * @param case true if the sort have lowercase values going first, false otherwise
+     *
+     * @updates this
+     *
+     * @ensures this = #this
```

Comment:
These ensures don't seem quite right. Is this saying that these methods don't do anything?